runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/client/AbstractWindmillStream.java

```diff
@@ -238,7 +239,7 @@ public final void appendSummaryHtml(PrintWriter writer) {
   protected abstract void appendSpecificHtml(PrintWriter writer);
 
   @Override
-  public final synchronized void close() {
+  public synchronized void close() {
```

Comment:
can this stay final?

Reply:
override is in GetDataStream since we want to clear all of the batches to prevent dangling calls